deploy/osps/default/osp-flatcar.yaml

```diff
@@ -849,7 +849,7 @@ spec:
         content:
           inline:
             data: |
-              {{- if .NetworkConfig }}
+              {{- if and (.NetworkConfig) (or (.NetworkConfig.CIDR) (.NetworkConfig.Gateway) (.NetworkConfig.DNS.Servers)) }}
```

Comment:
I don't think being explicit here instead of using `IsStaticIPConfig` is buying us all that much. 

Besides we have to be careful to be in sync with implementation of `IsStaticIPConfig` which we get for free if we directly use the function.

Reply:
As I explained earlier as well, we try to make these configurations as explicit to the user as possible. In case we use some internal method `IsStaticIPConfig` that's not even defined within the OSM base(comes from MC dep) it makes it ambiguous for the end users.

Reply:
> As I explained earlier as well,...

Yes. My comment is reply to that. 

> some internal method IsStaticIPConfig that's not even defined within the OSM codebase(comes from MC dep) it makes it ambiguous for the end users.

The me method is already in use in OSM. https://github.com/kubermatic/operating-system-manager/blob/457a0e085ff5cb1260a314b16bba32bb948fbf24/pkg/controllers/osc/resources/operating_system_config.go#L213

How does not defining it in OSM codebase make it ambiguous?

Reply:
> Yes. My comment is reply to that.

It's not really a reply to that Pratik. 

> The me method is already in use in OSM.

Yes exactly, that's maybe something that is confusing you. OSPs are not exactly in the "code base" but shipped with OSM. Users should be concerned with OSPs and not the code base.

Reply:
Things should be very descriptive and as far way as possible from any MC related functionality. We are trying to decouple OSM from MC or even from Kubermatic as possible as we can, as we would like to make it cluster-api like soon.